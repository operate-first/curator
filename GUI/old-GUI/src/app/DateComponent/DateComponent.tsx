import React from 'react';
import { CalendarAltIcon } from '@patternfly/react-icons';
import { InputGroup, InputGroupText, TextInput, FormGroup } from '@patternfly/react-core';

type myProps = {
  changeDate: Function;
  dateType: string;
  initialDate?: Date; // for unit testing/snapshots
}

type myState = {
  currentDate: Date;
}

class SimpleInputGroups extends React.Component<myProps, myState> {
  constructor(myProps: myProps) {
    super(myProps);

    this.state = {
      currentDate: myProps.initialDate || new Date()
    };
  }

  formatDate(date) {
    var d = new Date(date);
    var month = '' + (d.getMonth() + 1),
      day = '' + d.getDate(),
      year = d.getFullYear();

    if (month.length < 2)
      month = '0' + month;
    if (day.length < 2)
      day = '0' + day;

    return [year, month, day].join('-');
  }

  updateDateValue(newDate) {
    this.setState({ currentDate: newDate });
  }

  render() {
    return (
      <React.Fragment>
        <FormGroup label={this.props.dateType}
          isRequired
          fieldId={this.props.dateType}
        >
          <InputGroup>

            <TextInput
              name="textInput"
              id={this.props.dateType}
              type="date"
              aria-label="Input Date"
              onChange={value => { this.props.changeDate(value); this.updateDateValue(value) }}
              value={this.formatDate(this.state.currentDate)}
            />

            <InputGroupText component="label" htmlFor="textInput9">
              <CalendarAltIcon />
            </InputGroupText>

          </InputGroup>

        </FormGroup>

      </React.Fragment>
    );
  }
}

export { SimpleInputGroups };
