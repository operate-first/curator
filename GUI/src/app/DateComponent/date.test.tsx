import React from 'react';
import { shallow } from 'enzyme';
import { SimpleInputGroups as DateComponent } from './DateComponent';

describe('DateComponent tests', () => {

  test('should render DateComponent component', () => {

    const dropdown = shallow(<DateComponent changeDate={() => { }} dateType='StartDate' initialDate={new Date(946750000000)}/>);
    expect(dropdown).toMatchSnapshot();
  });

  test('should handle DateComponent change', () => {
    // eslint-disable-next-line no-undef
    const callback = jest.fn();
    const dropdown = shallow(<DateComponent changeDate={callback} dateType='StartDate' />);

    // simulate user changing value of component
    dropdown.find('#StartDate').simulate('change', 3);

    expect(callback.mock.calls.length).toBe(1);
    expect(callback.mock.calls[0][0]).toBe(3); // callback should be called with new value
  })

})
