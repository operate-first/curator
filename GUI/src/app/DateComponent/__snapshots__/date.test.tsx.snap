// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateComponent tests should render DateComponent component 1`] = `
<Fragment>
  <Component
    fieldId="StartDate"
    isRequired={true}
    label="StartDate"
  >
    <Component>
      <ForwardRef
        aria-label="Input Date"
        id="StartDate"
        name="textInput"
        onChange={[Function]}
        type="date"
        value="2000-01-01"
      />
      <Component
        component="label"
        htmlFor="textInput9"
      >
        <CalendarAltIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </Component>
    </Component>
  </Component>
</Fragment>
`;
